lib/howl/interactions/location_selection.moon

```diff
@@ -0,0 +1,46 @@
+-- Copyright 2012-2015 The Howl Developers
+-- License: MIT (see LICENSE.md at the top-level directory of the distribution)
+
+import app, interact, mode, Buffer from howl
+import highlight from howl.ui
+import Matcher from howl.util
+
+get_buffer_for_file = (file, local_buffers) ->
+  for buffer in *app.buffers
+    return buffer if buffer.file == file
+
+  for buffer in *local_buffers
+    return buffer if buffer.file == file
+
+  buffer = Buffer mode.for_file file
+  buffer.file = file
+  buffer.title = 'Preview: '..buffer.title
+  table.insert local_buffers, buffer
+  return buffer
+
+interact.register
+  name: 'select_location'
+  description: 'Selection list for locations - a location consists of a file (or buffer) and line number'
+  handler: (opts) ->
+    opts = moon.copy opts
+    editor = opts.editor or app.editor
+    buffer = editor.buffer
+    orig_buffer = buffer
+    local_buffers = {}
+
+    on_selection_change = opts.on_selection_change
+    opts.on_selection_change = (selection, text, items) ->
+      if selection
+        buffer = selection.buffer or get_buffer_for_file selection.file, local_buffers
+        editor.buffer = buffer
+        if selection.line_nr
+          editor.line_at_center = selection.line_nr
+
+      if on_selection_change
+        on_selection_change selection, text, items
+
+    result = interact.select opts
+    return result if result
```

Comment:
This will leave the temporary preview buffer in the editor? That doesn't seem like it's a good idea. I haven't seen this be a problem currently when I'm testing, but I think that's because the commands now using the interaction will replace it immediately anyway. But consider a command that is just going to use the result for something else. We'll end up with a preview buffer that will have to be manually closed.